examples/02_intermediate/wikimedia.ipynb

```diff
@@ -0,0 +1,513 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7bc632d9-b47c-4f3d-87fc-3364f90dab4c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datetime import datetime, timezone\n",
+    "\n",
+    "start_time = datetime(2024, 5, 8, 0, 0, 0)\n",
+    "end_time = datetime(2024, 5, 8, 23, 0, 0, tzinfo=timezone.utc)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "80702f29-a9c7-45e8-8e77-7cc717f4858a",
+   "metadata": {},
+   "source": [
+    "To read this data, we will use a [Pull input adapter](https://github.com/Point72/csp/wiki/Write-Historical-Input-Adapters#pullinputadapter). However, since our historical data can be downloaded into a csv file, we will use the existing implementation of a [`CSVReader` adapter](https://github.com/Point72/csp/blob/main/csp/adapters/csv.py) from CSP."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b513164c-9e2f-44c9-a7c2-9a93357f4cb9",
+   "metadata": {},
+   "source": [
+    "Again, we will use a Perspective widget to display the results of our historical data. This time, the widget will immediately show the historical data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "555b24f7-f765-4711-90a9-46db5555a50f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "from sseclient import SSEClient as EventSource, Event\n",
+    "from perspective import PerspectiveWidget, Table, Plugin\n",
+    "import pandas as pd\n",
+    "\n",
+    "start_date = datetime.isoformat(start_time)\n",
+    "end_date = datetime.isoformat(end_time)\n",
+    "\n",
+    "URL = f'https://stream.wikimedia.org/v2/stream/recentchange?since={start_date}'\n",
+    "\n",
+    "# Initial data\n",
+    "data = pd.DataFrame({\n",
+    "    \"servername\": [\"\"],\n",
+    "    \"time\": [datetime.utcnow()],\n",
+    "    \"event\": [\"\"],\n",
+    "    \"servername_count\": [0],\n",
+    "})\n",
+    "\n",
+    "raw_data = pd.DataFrame(dict([]))\n",
+    "table = Table(data, index=\"servername\")\n",
+    "servernames = dict([])\n",
+    "for event in EventSource(URL):\n",
+    "    if event.event == 'message':\n",
+    "        try:\n",
+    "            change = json.loads(event.data)\n",
+    "        except ValueError:\n",
+    "            pass\n",
+    "        else:\n",
+    "            # discard canary events\n",
+    "            # WMF Data Engineering team produces artificial 'canary' events into \n",
+    "            # each stream multiple times an hour. The presence of these canary\n",
+    "            # events in a stream allow us to differentiate between a broken event\n",
+    "            # stream, and an empty one.\n",
+    "            if change['meta']['domain'] == 'canary':\n",
+    "                continue\n",
+    "            if datetime.fromisoformat(change['meta']['dt']) > datetime.fromisoformat(end_date):\n",
```

Comment:
I'm unable to run the historical example in Python 3.8, I get the following error:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-80a0bc686bde> in <cell line: 22>()
     34             if change['meta']['domain'] == 'canary':
     35                 continue
---> 36             if datetime.fromisoformat(change['meta']['dt']) > datetime.fromisoformat(end_date):
     37                 print(f"Change at {datetime.fromisoformat(change['meta']['dt'])}:: stop")
     38                 break

ValueError: Invalid isoformat string: '2024-05-08T00:20:19Z'
```

Let's make sure we can run these examples with all `csp` Python versions, and if not, let's at least specify what Python version the example supports.

Reply:
Makes sense, I'll check thanks!